Documentation/writing-principles-and-guidelines/spelling-and-typo3-terminology.rst

```diff
@@ -1,111 +1,1006 @@
+        hidden in the TYPO3 backend for specific users or user groups. This is
+        done via the permission setup.
+
+    ..  _extbase:
+    -   *Extbase*
+
+        Extbase is a framework for developing extensions in the TYPO3 system
+        based on the Model-View-Controller (MVC) principle. It allows to easily
+        define models with their data fields, that can be managed by editors in
+        the backend. The models can then also be revealed in the frontend with
+        any custom logic, adhering to standards, conventions and API
+        definitions. The Extbase controllers which are part of an Extbase plugin
+        allow to integrate this custom logic in PHP.
+
+    ..  _extension:
+    -   *Extension*
+
+        An extension is an add-on to the TYPO3 system that adds additional
+        functionalities and features. An extension can consist of multiple
+        parts, for example backend modules, frontend plugins, scheduler tasks,
+        console commands, API definitions, frontend styling and much more.
+
+    ..  _extension-builder:
+    -   *Extension Builder*
+
+        The Extension Builder is a backend module in TYPO3 that facilitates the
+        creation of (preferably) extbase extensions. The Extension Builder is a
+        community extension and maintained on its own.
+
+    ..  _extension-manager:
+    -   *Extension Manager*
+
+        The Extension Manager is an interface in the TYPO3 backend used for
+        installing, updating, and managing extensions. It is very important in
+        legacy installations, but in Composer-based installations it is only
+        used for configuring extensions (composer then manages the
+        (de-)installation of extensions).
+
+    ..  _extension-scanner:
+    -   *Extension Scanner*
+
+        The Extension Scanner analyzes installed extensions for compatibility
+        issues for current and future TYPO3 versions. It can report fixes that
+        are needed to upgrade a custom extension.
+
+*   F
+
+    ..  _fal:
+    -   *FAL*
+
+        The File Abstraction Layer (FAL) is a system in TYPO3 that centralizes
+        the management and access to files and media resources. This is the
+        technical interface (API) to the integrated media asset database.
+
+    ..  _fe-groups---be-groups:
+    -   *fe_groups / be_groups*
+
+        Frontend groups :sql:`fe_groups` and backend groups :sql:`be_groups`
+        are user groups in TYPO3 that define different permissions and roles.
+
+    ..  _felogin:
+    -   *felogin*
+
+        :t3ext:`felogin` is a TYPO3 system extension for managing and
+        authenticating frontend users.
+
+    ..  _fe-users---be-users:
+    -   *fe_users / be_users*
+
+        Frontend users :sql:`fe_users` and backend users :sql:`be_users` are the
+        two main types of users in the TYPO3 system.
+
+    ..  _file-reference:
+    -   *file reference*
+
+        A file reference is a reference to a file stored and used within the
```

Comment:
file references reducing the webspace a lot as files related to multiple content elements and records are now set in relation by reference records and were not needed to copy anymore.